### PR TITLE
fix(jsx): Break opening elements with single text attribute if there's a comment

### LINKED ISSUE #3152: Inline comments inside jsx tags is broken.
**Prettier 1.7.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAGgHSgAhwenxwEMYYAnbAPm1X3SoG5tatcSzL3COLrb6TFlAzY8pCgF5gARgC+YgkQlca7eAA8YAhsyitFKyZhAnFPFf3VwtOofpFtxnY6dPdlnQy7Pu1eTW0RQUYQABoQCAAHGABLaABnZFBiBJhkADNiABsEuAiAI3JiMABrOBgAZSiS2KgAc2QKAFd8kAATCDBMnLyIgCsEjQAhYrKKyuIAWzgAGTq4Hty2iGaYKLWAJiW+kBryPPJkEALiAoBPbOhwvfI6mAB1WPaYAAtkAA4ABgi8qdimuRWr86vVsnAAIrNCDwHZtGBnJ4vd5ITYRCjEWLZUEAYQgUymxGOUGgiwizTyABUzkkkFllnI5EA)

**Input:**
```jsx
<X
  // attr
>
</X>;

<X
  attr
  // attr
>
</X>;

<X
  attr={1}
  // attr
>
  text
</X>;

<X
  attr=""
  // attr
>
  text
</X>;

<X
  attr=""
  // attr
  attr=""
>
  text
</X>;
```

**Output:**
```jsx
<X
// attr
/>;

<X
  attr
  // attr
/>;

<X
  attr={1}
  // attr
>
  text
</X>;

<X attr="">
// attr
  text
</X>;

<X
  attr=""
  // attr
  attr=""
>
  text
</X>;
```

**Expected behavior:**
```jsx
<X
  // attr // <- wrong indent
/>;

<X
  attr
  // attr
/>;

<X
  attr={1}
  // attr
>
  text
</X>;

<X
  attr=""
  // attr // <- bug: inline comments should remain here
>
  text
</X>;

<X
  attr=""
  // attr
  attr=""
>
  text
</X>;
```

### PR DESCRIPTION
**Pull request by duailibe**
Fix #3152 

**Input:**
```jsx
<div
  className="foo" // comment
>
  {children}
</div>;

<div
  className="foo"
  // comment
>
  {children}
</div>;
```

**Current in master:**
```jsx
<div className="foo"> // comment
  {children}
</div>;

<div className="foo">
// comment
  {children}
</div>;
```

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -1770,7 +1770,8 @@ function genericPrintNoParens(path, options, print, args) {
       if (
         n.attributes.length === 1 &&
         n.attributes[0].value &&
-        isStringLiteral(n.attributes[0].value)
+        isStringLiteral(n.attributes[0].value) &&
+        !n.attributes[0].comments
```
Comment (azz) [inline src/printer.js]: Can the attribute node have `trailingComments`?

Comment (duailibe) [reply src/printer.js]: Not sure but I don't think so.. it has a comment with `trailing: true`, but no `trailingComments`.

**Comment by duailibe** [comment]
Just caught another bug:

***Input:***
```jsx
<div // comment
  className="foo"
>
  {children}
</div>
```

***Master:***

```jsx
<div className="foo"> // comment
  {children}
</div>

File: src/printer.js (hunk 1 above)
Comment (duailibe) [reply src/printer.js]: Actually I don't see `trailingComments` anywhere on the code..

**Comment by duailibe** [comment]
@azz I think this covers all the cases now

**Pull request merged**